src/utils/tree.mli

```diff
@@ -0,0 +1,20 @@
+type 'a tree = { node : 'a; children : 'a forest }
+and 'a forest = 'a tree list
+
+val leaf : 'a -> 'a tree
+
+module type S = sig
+  type 'a t
+
+  val fold_left : f:('acc -> 'a -> 'acc) -> 'acc -> 'a t -> 'acc
+  val iter : f:('a -> unit) -> 'a t -> unit
+  val map : f:('a -> 'b) -> 'a t -> 'b t
+end
```

Comment:
Why not include `filter_map` in the signature, since it is defined for both trees and forests?

Reply:
Because the signature was different for the trees, and sknce it wasn’t used in the original PR I didn’t spend time trying to fit the same signature

Reply:
Indeed, filter_map on a tree may return None if the function decides to remove it. Fine not to include it.